src/app/GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -620,9 +620,42 @@ await DiffText.ViewChangesAsync(DiffFiles.SelectedItem,
         private void ShowSelectedFile(bool ensureNoSwitchToFilter = false, int? line = null)
         {
             DiffText.InvokeAndForget(() =>
-                blameToolStripMenuItem.Checked
-                    ? ShowSelectedFileBlameAsync(ensureNoSwitchToFilter, line)
-                    : ShowSelectedFileDiffAsync(ensureNoSwitchToFilter, line));
+                DiffFiles.SelectedFolder is RelativePath relativePath
+                    ? ShowSelectedFolderAsync(relativePath)
+                    : blameToolStripMenuItem.Checked
+                        ? ShowSelectedFileBlameAsync(ensureNoSwitchToFilter, line)
+                        : ShowSelectedFileDiffAsync(ensureNoSwitchToFilter, line));
+        }
+
+        private Task ShowSelectedFolderAsync(RelativePath relativePath)
+        {
+            (string path, string description) = GetDescription(relativePath, DiffFiles.SelectedItems.ToArray());
+            BlameControl.Visible = false;
+            DiffText.Visible = true;
+            return DiffText.ViewTextAsync(fileName: path, text: description);
+
+            static (string Path, string Text) GetDescription(RelativePath relativePath, FileStatusItem[] items)
+            {
+                string path = relativePath.Value;
+                int nameStartIndex = path.Length;
+                if (!path.EndsWith(PathUtil.PosixDirectorySeparatorChar))
+                {
+                    path += PathUtil.PosixDirectorySeparatorChar;
+                    if (path.Length > 1)
+                    {
+                        ++nameStartIndex;
+                    }
+                }
+
+                StringBuilder description = new();
+                description.Append('(').Append(items.Length).Append(") ").AppendLine(path);
```

Comment:
?
```suggestion
                description.Append($"({items.Length}) {path}");
```

Reply:
This may be shorter and easier to read. But there is a StringBuilder already, and I wanted to avoid building an interpolated string in addition.